src/frontend/org/voltcore/messaging/HostCriteria.java

```
@@ -0,0 +1,253 @@
+        checkArgument(startAction != null, "start action is null");
+        checkArgument(hostCount > 0, "host count %s is less then one", hostCount);
+
+        this.paused = paused;
+        this.configHash = configHash;
+        this.meshHash = meshHash;
+        this.enterprise = enterprise;
+        this.startAction = startAction;
+        this.bare = bare;
+        this.hostCount = hostCount;
+        this.nodeState = nodeState;
+        this.addAllowed = addAllowed;
+    }
+
+    public boolean isPaused() {
+        return paused;
+    }
+
+    public UUID getConfigHash() {
+        return configHash;
+    }
+
+    public UUID getMeshHash() {
+        return meshHash;
+    }
+
+    public boolean isEnterprise() {
+        return enterprise;
+    }
+
+    public StartAction getStartAction() {
+        return startAction;
+    }
+
+    public boolean isBare() {
+        return bare;
+    }
+
+    public int getHostCount() {
+        return hostCount;
+    }
+
+    public NodeState getNodeState() {
+        return nodeState;
+    }
+
+    public boolean isAddAllowed() {
+        return addAllowed;
+    }
+
+    public JSONObject appendTo(JSONObject jo) {
+        checkArgument(jo != null, "json object is null");
+        try {
+            jo.put(IS_BARE, bare);
+            jo.put(IS_ENTERPRISE, enterprise);
+            jo.put(IS_PAUSED, paused);
+            jo.put(CONFIG_HASH, configHash.toString());
+            jo.put(MESH_HASH, meshHash.toString());
+            jo.put(START_ACTION, startAction.name());
+            jo.put(HOST_COUNT, hostCount);
+            jo.put(NODE_STATE, nodeState.name());
+            jo.put(ADD_ALLOWED, addAllowed);
+        } catch (JSONException e) {
+            Throwables.propagate(e);
+        }
+        return jo;
+    }
+
+    public boolean isUndefined() {
+        return UNDEFINED.equals(configHash) && UNDEFINED.equals(meshHash);
+    }
+
+    @Override
+    public int hashCode() {
```

Is there a good reason this needs to be implemented? Does anyone call this? Can we just make it throw unsupported?